### PR TITLE
⚠️ Use Kubernetes 1.26 in Quick Start docs and CAPD.

### PR DESCRIPTION
**Pull request by aniruddha2000**
Signed-off-by: Aniruddha Basak <codewithaniruddha@gmail.com>

**What this PR does / why we need it**:

Updates the Quick Start guide to use Kubernetes 1.26.
Part of #7671

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

**Review by ykakarap — COMMENTED** [review]
Did a first pass review. Looks good.
Will verify quick-start and report back.

**Review by alexander-demicev — COMMENTED** [review]
seems fine to me

File: test/e2e/config/docker.yaml [hunk 1]
```diff
@@ -254,10 +254,10 @@ variables:
...
   # This avoids building node images in the default case which improves the test duration significantly.
-  KUBERNETES_VERSION_MANAGEMENT: "v1.25.3"
-  KUBERNETES_VERSION: "v1.25.3"
-  KUBERNETES_VERSION_UPGRADE_FROM: "v1.24.7"
-  KUBERNETES_VERSION_UPGRADE_TO: "v1.25.3"
+  KUBERNETES_VERSION_MANAGEMENT: "v1.26.0"
+  KUBERNETES_VERSION: "v1.26.0"
+  KUBERNETES_VERSION_UPGRADE_FROM: "v1.25.3"
+  KUBERNETES_VERSION_UPGRADE_TO: "v1.26.0"
   ETCD_VERSION_UPGRADE_TO: "3.5.4-0"
```
Comment (sbueringer) [inline test/e2e/config/docker.yaml:261]: Looks like kubeadm is now using `3.5.6-0`, let's update the var here accordingly

(https://github.com/kubernetes/kubernetes/blob/v1.26.0/cmd/kubeadm/app/constants/constants.go#L312)

**Comment by sbueringer** [comment]
Thx! one nit + "Will verify quick-start and report back." from @ykakarap

**Comment by nawazkh** [comment]
looks good to me!

**Comment by sbueringer** [comment]
@aniruddha2000 Can you please rebase?

File: test/e2e/config/docker.yaml [hunk 2]
```diff
@@ -254,11 +254,11 @@ variables:
...
-  KUBERNETES_VERSION: "v1.25.3"
-  KUBERNETES_VERSION_UPGRADE_FROM: "v1.24.7"
-  KUBERNETES_VERSION_UPGRADE_TO: "v1.25.3"
-  ETCD_VERSION_UPGRADE_TO: "3.5.4-0"
+  KUBERNETES_VERSION_MANAGEMENT: "v1.26.0"
+  KUBERNETES_VERSION: "v1.26.0"
+  KUBERNETES_VERSION_UPGRADE_FROM: "v1.25.3"
+  KUBERNETES_VERSION_UPGRADE_TO: "v1.26.0"
+  ETCD_VERSION_UPGRADE_TO: "3.5.6-0"
   COREDNS_VERSION_UPGRADE_TO: "v1.9.3"
```
Comment (sbueringer) [inline test/e2e/config/docker.yaml:262]: Can you please also bump v1.25.3 => v1.26.0 in clusterctl_upgrade_test.go?

Comment (sbueringer) [reply test/e2e/config/docker.yaml:262]: Please also add a bullet point to "Modify quickstart and CAPD to use the new Kubernetes release:" in `kubernetes_bump.md`. Something like "bump `InitWithKubernetesVersion` in `clusterctl_upgrade_test.go`

**Review by sbueringer — COMMENTED** [review]
Rebase + last nits otherwise I think we can merge

File: test/e2e/config/docker.yaml (hunk 2 above)
Comment (aniruddha2000) [reply test/e2e/config/docker.yaml:262]: Done

**Comment by aniruddha2000** [comment]
Sure!

**Comment by sbueringer** [comment]
/test pull-cluster-api-e2e-full-main
/test pull-cluster-api-e2e-workload-upgrade-1-25-latest-main

/lgtm

Will approve after successful tests.

**Comment by sbueringer** [comment]
/approve

Thank you! :tada:

**Comment by sbueringer** [comment]
@aniruddha2000 Can you please open manual cherry-picks for release-1.3 and release-1.2?

**Pull request merged**